projects/rocprofiler-sdk/source/lib/rocprofiler-sdk/hip/stream.cpp

```diff
@@ -75,14 +75,28 @@ get_stream_map()
     return _v;
 }
 
+auto&
+get_idx_offset()
```

Comment:
While I'll concede that, in this current implementation, it appears to be thread-safe, I want to note this is a dangerous design... it will be very easy for future developers to unknowingly access this value in a thread-unsafe way